Wikipedia/Code/SavedArticlesFetcher.m

```diff
@@ -206,10 +213,15 @@ - (AnyPromise*)fetchImageInfoForImagesInArticle:(MWKArticle*)article {
 
     if (imageFileTitles.count == 0) {
         DDLogVerbose(@"No image info to fetch, returning successful promise with empty array.");
-        return [AnyPromise promiseWithValue:imageFileTitles];
+        success(imageFileTitles);
+        return;
     }
 
-    return PMKJoin([[imageFileTitles bk_map:^AnyPromise*(NSString* canonicalFilename) {
+    for (NSString *canonicalFilename in imageFileTitles) {
+        [self.imageInfoFetcher fetchGalleryInfoForImage:canonicalFilename fromSite:article.title.site];
```

Comment:
Hey just a quick check... The gallery should almost never be fetching the canonical images - these can be tens if not hundreds of MB. iirc we formerly were fetching *up to* either 1024 or 1280 width images for gallery.